src/core/externalstorage/qgshttpexternalstorage.cpp

```diff
@@ -192,14 +209,49 @@ QString QgsWebDAVExternalStorage::displayName() const
 
 QgsExternalStorageStoredContent *QgsWebDAVExternalStorage::doStore( const QString &filePath, const QString &url, const QString &authcfg ) const
 {
-  return new QgsWebDAVExternalStorageStoredContent( filePath, url, authcfg );
+  return new QgsHTTPExternalStorageStoredContent( filePath, url, authcfg );
 };
 
 QgsExternalStorageFetchedContent *QgsWebDAVExternalStorage::doFetch( const QString &url, const QString &authConfig ) const
 {
   QgsFetchedContent *fetchedContent = QgsApplication::networkContentFetcherRegistry()->fetch( url, Qgis::ActionStart::Deferred, authConfig );
 
-  return new QgsWebDAVExternalStorageFetchedContent( fetchedContent );
+  return new QgsHTTPExternalStorageFetchedContent( fetchedContent );
 }
 
+
+// AWS S3 PROTOCOL
+
+QString QgsAWSS3ExternalStorage::type() const
+{
+  return QStringLiteral( "AWSS3" );
+};
+
+QString QgsAWSS3ExternalStorage::displayName() const
+{
+  return QObject::tr( "AWS S3" );
+};
+
+QgsExternalStorageStoredContent *QgsAWSS3ExternalStorage::doStore( const QString &filePath, const QString &url, const QString &authcfg ) const
+{
+  QgsHTTPExternalStorageStoredContent* storedContent = new QgsHTTPExternalStorageStoredContent( filePath, url, authcfg );
```

Comment:
It is probably better to use a std::unique_ptr and release it when returning.
We try to avoid new and delete when possible.